extmod/modframebuf.c

```diff
@@ -115,7 +171,7 @@ static inline uint32_t getpixel(const mp_obj_framebuf_t *fb, int x, int y) {
 }
 
 STATIC void fill_rect(const mp_obj_framebuf_t *fb, int x, int y, int w, int h, uint32_t col) {
-    if (x + w <= 0 || y + h <= 0 || y >= fb->height || x >= fb->width) {
```

Comment:
asking is always a good thing ) and actually fill_rect is a related part to gray4 fill rect ) I will return previous check. But as I see it - there is a need to check for zero height/width too. No need to call subroutines if there is nothing to do